components/motor/gpio/motor_encoder.go

```diff
@@ -458,7 +460,14 @@ func (m *EncodedMotor) GoTo(ctx context.Context, rpm, targetPosition float64, ex
 
 // ResetZeroPosition sets the current position (+/- offset) to be the new zero (home) position.
 func (m *EncodedMotor) ResetZeroPosition(ctx context.Context, offset float64, extra map[string]interface{}) error {
-	return m.encoder.ResetPosition(ctx, extra)
+	if err := m.Stop(ctx, extra); err != nil {
+		return err
+	}
+	if err := m.encoder.ResetPosition(ctx, extra); err != nil {
+		return err
+	}
+	m.offsetInTicks = -1 * offset * m.ticksPerRotation
```

Comment:
Does this ever race with reconfigure? DO we need to have it protected by a lock too?

Reply:
https://github.com/viamrobotics/rdk/actions/runs/6647720265/job/18063608019 exposed a read-write race with https://github.com/viamrobotics/rdk/pull/3083/files#diff-c995a39b5b86039be0b4c03ff28ef502f44068848c99b80628eeaa9c9cef734dR481

Reply:
Sigh .... I'll pick up the ticket, at least I know its not with Reconfigure because this component is always rebuild 😄

Reply:
Thanks Rand! I've assigned [the ticket](https://viam.atlassian.net/browse/RSDK-5508). Had to fix the bot around identifying/ticketing data races.